akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -293,7 +295,12 @@ object ShardRegion {
    */
   def getRegionStatsInstance = GetShardRegionStats
 
-  @SerialVersionUID(1L) final case class ShardRegionStats(stats: Map[ShardId, Int])
+  /**
+   *
+   * @param stats the region stats mapping of `ShardId` to number of entities
+   * @param failed set of shards if any failed to respond within the timeout
+   */
+  @SerialVersionUID(1L) final case class ShardRegionStats(stats: Map[ShardId, Int], failed: Set[ShardId])
```

Comment:
I wrote it somewhere else already but to have it here as well: might be hard to extend it in a binary compatible way, so an easier way out might be to introduce a new request/response model and make the response a trait (that is only privately implemented) so that it can later be extended with more information (it seemed that was the original idea).

Reply:
LMK what you think on the updated to class commit? I like it.